docs(treesitter): fix grammar and clarity in query syntax note (#31477)
neovim/neovim: runtime/doc/treesitter.txt

Queries are written in a lisp-like language documented in
https://tree-sitter.github.io/tree-sitter/using-parsers#query-syntax

> Note: The predicates listed on that page differ from the ones Nvim supports.
> See |treesitter-predicates| for a complete list of predicates supported by
> Nvim.

Reviewer: ```suggestion
Note: The predicates listed there differ from those Nvim supports. See
|treesitter-predicates| for a complete list of predicates supported by Nvim.
```
